src/susceptibility.cpp

```diff
@@ -208,24 +208,30 @@ void lorentzian_susceptibility::update_P
 	SWAP(const realnum *, s1, s2);
       }
       if (s1 && s2) { // 3x3 anisotropic
-	LOOP_OVER_VOL_OWNED(gv, c, i) {
+	LOOP_OVER_VOL_OWNED(gv, c, i) if(s[i]!=0) {
+		//"if" to avoid instabilities coming from yee-grid shifts of off-diagnal elements
++		//FIXME: this prevents materials with pure off-diagonal sigma from working
```

Comment:
Wouldn't a purely off-diagonal sigma matrix be indefinite, and hence lead to both loss and gain?